public/components/tooltip/create_tooltip.tsx

```diff
@@ -25,6 +25,12 @@ export function isTooltipEnabledLayer(
   );
 }
 
+export function isTooltipEnabledOnHover(
+  layer: MapLayerSpecification
+): layer is DocumentLayerSpecification {
+  return isTooltipEnabledLayer(layer) && !layer.source.disableTooltipsOnHover;
```

Comment:
```suggestion
    if (!isTooltipEnabledLayer(layer)) {
    return false;
  }

  return !layer.source?.disableTooltipsOnHover;
```

Reply:
why?

Reply:
I think this way the logic can be more clear, and adding `?` can prevent errors if the source property is missing.

Reply:
source is not optional parameters in DocumentLayerSpecification. I assume typescript would have complained if '?' is missing. TBH, suggestion looks very verbose.

Reply:
i made this parameter is optional and added '?'

Reply:
Thanks on the change, I think the splitting or not is okay, IMO splitting is easier for reading the code